core/rpc/src/lib.rs

```diff
@@ -101,21 +96,77 @@ impl<C: Collection> Rpc<C> {
 #[async_trait::async_trait]
 impl<C: Collection> WithStartAndShutdown for Rpc<C> {
     async fn start(&self) {
-        let server = self
-            .build_server_from_config()
-            .await
-            .expect("RPC Server to build");
+        let (stop, server_handle) = stop_channel();
+        let json_rpc_service = JSONRPCServer::builder()
+            .to_service_builder()
+            .build(Methods::from(self.module.clone()), stop.clone());
+
+        let make_service = make_service_fn(move |_| {
+            let json_rpc_service = json_rpc_service.clone();
+
+            async move {
+                Ok::<_, hyper::Error>(service_fn(move |req: hyper::Request<hyper::Body>| {
+                    let mut json_rpc_service = json_rpc_service.clone();
```

Comment:
I would actually like someone else to take a look at this:

Due to the bounds on FnMut, moving into the async clause gives a compiler error without the explicit clones there. Not super why or how to avoid this

The clone really isnt to expensive, but its still cloning like Vec<String>::len() > 50 per connection

Reply:
I think we can def wrap it in an `Arc` to make clone significantly cheaper, but still feels kinda extra

also make_serivce should not be a public api for http servers 😭😭